lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -72,6 +72,17 @@ WasmBinaryReader::ReadFromModule()
 
     SectionCode sectionId;
     UINT length = 0;
+
+    if (ReadConst<UINT32>() != 0x6d736100)
```

Comment:
this will not work: it will be called for every function. should have some check to make sure this is only executed first time you call